### ISSUE TITLE
try to add admin interface languages not working on v14.15.4

### ISSUE THREAD

**Issue by BasharOsman** [strapi/strapi#18786]
### Required System information

- Node.js version: 20.9
- NPM version: 10.1.0
- Strapi version: 14.15.4
- Database: sqlite
- Operating system: ubuntu
- Is your project Javascript or Typescript: JS

### Describe the bug

if you want to change the admin panel language for English to any language like French "fr" you need to change or add 
```javascript 
export default {
  config: {
    locales: ["fr"],
  },
};
```
  to src/admin/app.js
but did not found any language on the preferred language list on user profile

### Steps to reproduce the behavior

1. add language to locales on src/admin/app.js
2. build the strapi by using "strapi build" command
3. go to user admin profile preferred language list will have only English language 

### Expected behavior

there should be more languages than English after update the app.js file with locales needed

### Screenshots
...

### Code snippets
src/admin/app.js
```javascript 
export default {
  config: {
    locales: ["fr"],
  },
};
```

### Additional context

tested on v4.14.6 no problem

**Comment by Schlipak** [comment]
I'm currently running 14.15.0 and it works fine, but I'm having the same issue for any versions above that, I had it first happen in 14.15.1 (before that version got deprecated), but the same happens in 14.15.4.
In fact nothing in my app.js (or tsx in my case) gets executed, additional languages don't work, custom translation overrides are ignored, and the bootstrap code isn't run.

**Comment by joshuaellis** [comment]
this will be fixed in 4.15.5, we've already submitted a PR to solve the issue.
